src/components/studies/adherenceHooks.ts

```diff
@@ -73,13 +76,13 @@ export const useAdherenceAlerts = (
   studyId: string,
   categories: AdherenceAlertCategory[],
   pageSize: number = 25,
-  offsetBy = 0
+  currentPage = 0
 ) => {
-  const cat = categories.join()
+  const cat = categories.sort().join()
   const {token} = useUserSessionDataState()
   return useQuery<{items: AdherenceAlert[]; total: number}, ExtendedError>(
-    ADHERENCE_KEYS.alerts(studyId, cat, offsetBy, pageSize),
-    () => AdherenceService.getAdherenceAlerts(studyId, categories, pageSize, offsetBy, token!),
+    ADHERENCE_KEYS.alerts(studyId, cat, currentPage, pageSize),
+    () => AdherenceService.getAdherenceAlerts(studyId, categories, pageSize, currentPage, token!),
```

Comment:
FYI, I see the same bug that we just had to fix in Synapse. If the `token` changes, neither `invalidateQueries` nor `resetQueries` would result in the query function using the new token. 

We added the token to the query key, but a better solution might be to have the AdherenceService pull the token from a stateful store (which is what we will probably end up switching to in the future).

Reply:
@nickgros, this makes sense, but I don't think it would be an issue in MTB since I can't think of the scenario where the token would change without the app reinitializing, since login involves redirection away from the app itself